python/tests/mlir/ast_decrementing_range.py

```diff
@@ -0,0 +1,82 @@
+# ============================================================================ #
+# Copyright (c) 2022 - 2024 NVIDIA Corporation & Affiliates.                   #
+# All rights reserved.                                                         #
+#                                                                              #
+# This source code and the accompanying materials are made available under     #
+# the terms of the Apache License 2.0 which accompanies this distribution.     #
+# ============================================================================ #
+
+# RUN: PYTHONPATH=../../ pytest -rP  %s | FileCheck %s
+
+import os
+
+import pytest
+import numpy as np
+
+import cudaq
+
+
+def test_decrementing_range():
+
+    @cudaq.kernel(jit=True)
+    def test(q:int, p:int):
+        qubits = cudaq.qvector(5) 
+        for k in range(q, p, -1):
+            x(qubits[k])
+
+    print(test)
+    test(2, 0)
+
+# CHECK-LABEL:   func.func @__nvqpp__mlirgen__test(
+# CHECK-SAME:                                      %[[VAL_0:.*]]: i64,
+# CHECK-SAME:                                      %[[VAL_1:.*]]: i64) attributes {"cudaq-entrypoint"} {
```

Comment:
I think this case is fine, but as a heads-up, the script can get these wrong. For example, if there were 3 arguments of type `i64` we'd get back to back lines
```
# CHECK-SAME:                                      %[[VAL_0:.*]]: i64,
# CHECK-SAME:                                      %[[VAL_1:.*]]: i64,
```
That is a problem because of the greedy match on the first line of the pair.

In my experience, it is always a good idea to clean these up by fusing the CHECK-SAME lines back together. We can also erase tons of gratuitous whitespace that the script adds for no reason.